### PR TITLE
Change minRunners behavior and fix the new listener min runners

### LINKED ISSUE #2707: gha-runner-scale-set - pre-emptively scale runners based on new "min idle runners" setting
### What would you like added?

The new scaleset runners already support a ```minRunners``` and ```maxRunners``` setting in the gha-runner-scale-set helm charts:

https://github.com/actions/actions-runner-controller/blob/d134dee14b5ecd76ecbd2b3b2e1bc79fe3f4a95d/charts/gha-runner-scale-set/values.yaml#L39-L43

but it would be great if the runners also had a setting called something like ```minIdleRunners``` as well so that the runner group would scale up pre-emptively to ensure there were idle runners already available when a new workflow run arrives in the queue.

### Why is this needed?

Sometimes, scaling up can be a slow operation and waiting to do it on-demand when a new workflow is triggered could mean the workflow is delayed by several minutes while the scale-up event is processed - for example if a new node needs to be dynamically provisioned and added to the kubernetes cluster for the runner pod to be scheduled onto.

Scaling up pre-emptively to ensure there are always a predefined number of idle runners would mean workflows could start sooner.

For example, if the *current* gha-runner-scale-set ```values.yaml``` is configured as follows:

```
maxRunners: 10
minRunners: 2
```

... then the first two workflows that arrive will be immediately assigned to the two idle runners (lets assume they both take about 10-15 minutes to complete) . There are now 2 runners but they're both busy, so if a third workflow is triggered 5 minutes later it will need to wait for another runner pod to be provisioned and *that* could also require a new node to be dynamically provisioned and added to the kubernetes cluster, which might take several minutes.

If there were a ```minIdleRunners``` setting, the scale-up event would be triggered as soon as the two initial workflows were allocated to runners and became busy, so that the scale up to create 2 more *idle* runners would have already competed by the time the third workflow is triggered and a runner would be available to start the third workflow immediately.

### Additional context

**Note:** The new setting could potentially conflict with the existing ```minRunners``` and ```maxRunners``` in some cases - e.g.

```
maxRunners: 10
minRunners: 2
minIdleRunners: 100
```
or
```
maxRunners: 10
minRunners: 2
minIdleRunners: 0
```
or if there are, say, already 9 "busy" runners there could only be one additional idle runner provisioned.

The ```minIdleRunners``` setting would be "advisory" and would yield to the ```minRunners```and ```maxRunners``` settings when a conflict arises.

### PR DESCRIPTION
**Pull request by nikola-jokic**
This PR implements the [proposed ADR](https://github.com/actions/actions-runner-controller/blob/master/docs/adrs/2023-11-02-min-runners-strategy.md) and changes its status.

Target runner count is now calculated as `min(assigned + minRunners, maxRunners)` where maxRunners is set to the maximum number that can fit in int32 if not configured.

Also, this PR fixes a bug in the new listener that is not yet published, introduced by #3096

Fixes #2707

### CODE REVIEW

File: .github/workflows/gha-e2e-tests.yaml [hunk 1]
```diff
@@ -880,3 +880,98 @@ jobs:
...
+            if [ "$count" -ge 60 ]; then
+              echo "Timeout waiting for listener pod with label actions.github.com/scale-set-name=$ARC_NAME"
+              exit 1
+            fi
+            sleep 1
+            count=$((count+1))
+          done
+          kubectl wait --timeout=30s --for=condition=ready pod -n arc-systems -l actions.github.com/scale-set-name=$ARC_NAME
+          kubectl get pod -n arc-systems
+      - name: Ensure 5 runners are up
```
Comment (Link-) [inline .github/workflows/gha-e2e-tests.yaml:962]: I think this test is not sufficient, what I think the test should do is also to trigger one or more workflow runs and make sure we have 5 runners all the time

**Review by Link- — APPROVED** [review]
I don't want to delay the release for an e2e test. I'm fine with this moving forward as long as we address the e2e right after

**Comment by remover** [comment]
@nikola-jokic will this also apply to legacy ARC? i'm guessing not but would it be easy enough to apply there also?

**Comment by Link-** [comment]
@remover this does not apply to legacy ARC. Legacy ARC is maintained by the community led by @mumoshu and we do not backport changes.

**Comment by remover** [comment]
@mumoshu do you think this would be an easy enough change in legacy ARC? happy to help if so... :)

**Pull request merged**